contracts/imsv-funding/Master.algo.ts

```diff
@@ -663,13 +656,17 @@ export class Master extends Contract.extend(Ownable, Pausable) {
      *
      * @param mbr - The minimum balance requirement for storing the card fund data. Use `getCardFundMbr()`.
      * @param cardFundAddr - The address of the card fund account.
+     * @param ref - Client reference to store on the Card Fund
      * @returns Card fund address
      */
     @allow.call('NoOp')
     @allow.call('CloseOut')
-    cardFundDeployComplete(mbr: PayTxn, cardFundAddr: Address): Address {
+    cardFundDeployComplete(mbr: PayTxn, cardFundAddr: Address, ref: bytes64): Address {
         assert(this.cardFundsSetup(this.txn.sender, cardFundAddr).exists, 'CARD_FUND_NOT_FOUND');
 
+        // Check the card fund reference length
+        assert(ref.length <= MaxCardFundReferenceLength, 'CARD_FUND_REFERENCE_TOO_LONG');
+
```

Comment:
If `ref` is being defined as a fixed length argument, then the underlying generated TEAL will check the length during the call, so this can be removed.